test/extended/openstack/loadbalancers.go

```diff
@@ -382,6 +383,152 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
 			e2e.Logf("Pod %s successfully accessed through svc loadbalancer on ip %s and port %d", podName, fip.FloatingIP, svcPort)
 		})
 	}
+
+	// Only valid with the Octavia amphora provider. The OVN-Octavia provider does not
+	// support allowed_cidrs in the LB (https://issues.redhat.com/browse/OCPBUGS-2789)
+	g.It("should limit service access on an UDP Amphora LoadBalancer when an UDP LoadBalancer svc setting the loadBalancerSourceRanges spec is created on Openshift", func() {
+
+		// Skipped for OVN-Octavia provider
+		skipIfNotLbProvider("Amphora", cloudProviderConfig)
+
+		g.By("Creating Openshift deployment")
+		depName := "udp-lb-sourceranges-dep"
+		svcName := "udp-lb-sourceranges-svc"
+		svcPort := int32(8082)
+		replicas := int32(2)
+		labels := map[string]string{"app": depName}
+		drop_sourcerange := "9.9.9.9/32"
+		testDeployment := createTestDeployment(depName, labels, replicas, v1.ProtocolUDP, 8081)
+		deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(context.TODO(),
+			testDeployment, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By(fmt.Sprintf("Creating Openshift LoadBalancer Service with loadBalancerSourceRanges: '%s'", drop_sourcerange))
+		jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
+		jig.Labels = labels
+		svc, err := jig.CreateLoadBalancerService(loadBalancerServiceTimeout, func(svc *v1.Service) {
+			svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: svcPort, TargetPort: intstr.FromInt(8081)}}
+			svc.Spec.Selector = labels
+			svc.Spec.LoadBalancerSourceRanges = []string{drop_sourcerange}
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Checks from openshift perspective")
+		loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+		o.Expect(loadBalancerId).ShouldNot(o.BeEmpty(), "load-balancer-id annotation missing")
+		e2e.Logf("LB ID: %s", loadBalancerId)
+		o.Expect(svc.Spec.LoadBalancerSourceRanges).Should(o.Equal([]string{drop_sourcerange}))
```

Comment:
This one feels really redundant, I don't think we should check here if OpenShift API saves objects correctly.

Reply:
Agree, I've removed it